examples/DataApi-Voting-poll/template/modules/shared_data/src/main/scala/com/my/currency/shared_data/Main.scala

```diff
@@ -0,0 +1,360 @@
+    val validateSnapshotInterval = IO {
+      lastSnapshotOrdinal match {
+        case Some(value) => validatePollSnapshotInterval(value, state, update)
+        case None => ().validNec
+      }
+    }
+
+    val validatePoll = IO {
+      validateIfVotePollExists(state.polls.get(update.pollId))
+    }
+
+    val validateOption = IO {
+      validateIfOptionExists(state.polls.get(update.pollId), update.option)
+    }
+
+    val validateRepeatedVote = IO {
+      validateIfUserAlreadyVoted(state.polls.get(update.pollId), update.address)
+    }
+
+    for {
+      validatedSnapshotInterval <- validateSnapshotInterval
+      validatedPoll <- validatePoll
+      validatedOption <- validateOption
+      validatedRepeatedVote <- validateRepeatedVote
+    } yield validatedSnapshotInterval.productR(validatedPoll).productR(validatedOption).productR(validatedRepeatedVote)
+
+  }
+
+  def createPollValidationsWithSignature(update: CreatePollUpdate, proofs: NonEmptySet[SignatureProof], state: State)(implicit sp: SecurityProvider[IO]): IO[DataApplicationValidationErrorOr[Unit]] = {
+    val validateAddress = proofs
+      .map(_.id)
+      .toList
+      .traverse(_.toAddress[IO])
+      .map(validateProvidedAddress(_, update.owner))
+
+    val validations = createPollValidations(update, state, None)
+    for {
+      validatedAddress <- validateAddress
+      validatedPoll <- validations
+    } yield validatedAddress.productR(validatedPoll)
+  }
+
+  def voteInPollValidationsWithSignature(update: VoteInPollUpdate, proofs: NonEmptySet[SignatureProof], state: State)(implicit sp: SecurityProvider[IO]): IO[DataApplicationValidationErrorOr[Unit]] = {
+    val validateAddress = proofs
+      .map(_.id)
+      .toList
+      .traverse(_.toAddress[IO])
+      .map(validateProvidedAddress(_, update.address))
+
+    val validations = voteInPollValidations(update, state, None)
+
+    for {
+      validatedAddress <- validateAddress
+      validatedPoll <- validations
+    } yield validatedAddress.productR(validatedPoll)
+  }
+}
+
+object Combiners {
+  def combineCreatePoll(poll: CreatePollUpdate, acc: State): State = {
+    val pollId = Hash.fromBytes(customUpdateSerialization(poll)).toString
+    val pollOptions = poll.pollOptions.flatMap(option => Map(option -> 0L)).toMap
+    val newState = Poll(pollId, poll.name, poll.owner, pollOptions, Map.empty, poll.startSnapshotOrdinal, poll.endSnapshotOrdinal)
+
+    acc.focus(_.polls).modify(_.updated(pollId, newState))
+  }
+
+  def combineVoteInPoll(voteInPoll: VoteInPollUpdate, acc: State): State = {
+    val currentState = acc.polls(voteInPoll.pollId)
+    val currentOptionNumber = currentState.pollOptions(voteInPoll.option)
+
+    val newState = currentState
+      .focus(_.pollOptions)
+      .modify(_.updated(voteInPoll.option, currentOptionNumber + 1))
+      .focus(_.votes)
+      .modify(_.updated(voteInPoll.address, Map(voteInPoll.option -> 1)))
```

Comment:
It looks like this sets the `voteInPoll.option` value to 1. Shouldn't it add 1 to the current value?

Reply:
Oh sorry, the name is confusing, This only sets that the user voted in this option, currently, the user can only vote 1 in the option. The option above pollOptions that we're increasing in 1 is the part you mentioned